### PR TITLE
[queuing] clean the QueuingManager to not itself be a queue

### PR DESCRIPTION
**Pull request by timmysilv**
**Context:**
`QueuingManager` is itself a queue, and this makes talking about it rather confusing.

**Description of the Change:**
Replace `QueuingManager._active_contexts` (a list) with `QueuingManager._active_context` (a reference to a context), and also set `context._previous_context` to create a linked-list.

**Benefits:**
Code looks cleaner, talking about it is less confusing. Interacting with the active context would always involve indexing the last value of a list before, but now it just uses the saved reference.

**Possible Drawbacks:**
No real drawbacks. I suppose you could quickly index hidden contexts deep in the queue before this PR, but that was never part of the public API anyway (and I don't think it should be).

### CODE REVIEW

File: pennylane/queuing.py [hunk 1]
```diff
@@ -199,28 +199,31 @@ class QueuingManager:
...
 
-    _active_contexts = []
+    _active_context = None
     """The stack of contexts that are currently active."""
 
     @classmethod
     def add_active_queue(cls, queue):
         """Makes a queue the currently active recording context."""
-        cls._active_contexts.append(queue)
+        queue._previous_context = cls._active_context
```
Comment (albi3ro) [inline pennylane/queuing.py:208]: What happens if there was a previous previous context?

Comment (timmysilv) [reply pennylane/queuing.py:208]: it'll still have the attribute on there! I'm basically turning this python list of AnnotatedQueues into a linked-list

Comment (mudit2812) [reply pennylane/queuing.py:208]: Today I learned you can add attributes to instantiated objects in Python :)

**Review by mudit2812 — COMMENTED** [review]
Thanks for the changes Matt. Tests should probably be updated to check the linked-list structure of the active context works as expected. Since this doesn't impact the public API at all, I'm not sure if a changelog entry is necessary, but I'll leave this note here.

**Comment by timmysilv** [comment]
doesn't add anything, won't do

**Pull request closed without merging**